src/test_runner.cr

```diff
@@ -76,58 +68,68 @@ module Mint
+                    let test = this.suite.tests.shift()
 
-                  let currentHistory = window.history.length
+                    let currentHistory = window.history.length
 
-                  let result = await test.proc(this.suite.constants)
+                    let result = await test.proc(this.suite.constants)
 
-                  // Go back to the beginning
-                  if (window.history.length - currentHistory) {
-                    window.history.go(-(window.history.length - currentHistory))
-                  }
+                    // Go back to the beginning
+                    if (window.history.length - currentHistory) {
+                      window.history.go(-(window.history.length - currentHistory))
+                    }
 
-                  // Clear storages
-                  sessionStorage.clear()
-                  localStorage.clear()
+                    // Clear storages
+                    sessionStorage.clear()
+                    localStorage.clear()
 
-                  // TODO: Reset Stores
+                    // TODO: Reset Stores
 
-                  if (result instanceof TestContext) {
-                    try {
-                      await result.run()
-                      this.socket.send(JSON.stringify({ type: "SUCCEEDED", name: test.name, result: result.subject.toString() }))
-                    } catch (error) {
-                      this.socket.send(JSON.stringify({ type: "FAILED", name: test.name, result: error.toString() }))
-                    }
-                  } else {
-                    if (result) {
-                      this.socket.send(JSON.stringify({ type: "SUCCEEDED", name: test.name, result: "true" }))
+                    if (result instanceof TestContext) {
+                      try {
+                        await result.run()
+                        this.socket.send(JSON.stringify({ type: "SUCCEEDED", name: test.name, result: result.subject.toString() }))
+                      } catch (error) {
+                        this.socket.send(JSON.stringify({ type: "FAILED", name: test.name, result: error.toString() }))
+                      }
                     } else {
-                      this.socket.send(JSON.stringify({ type: "FAILED", name: test.name, result: "false" }))
+                      if (result) {
+                        this.socket.send(JSON.stringify({ type: "SUCCEEDED", name: test.name, result: "true" }))
+                      } else {
+                        this.socket.send(JSON.stringify({ type: "FAILED", name: test.name, result: "false" }))
+                      }
                     }
-                  }
 
-                  this.next(resolve, reject)
+                    this.next(resolve, reject)
+                  } catch (error) {
+                    // An error occurred while trying to run a test; this is different from the test itself failing.
+                    reject(error.toString());
```

Comment:
To signify erroring test I'd use sth like

```suggestion
                    this.socket.send(JSON.stringify({ type: "ERRORED", name: test.name, result: error.toString() }))
```

Reply:
Done; to make the test name accessible, I moved the `try` block to surround the `let result = ...` line and everything after that instead. I've edited the `TestRunner` class to accept `ERRORED` messages as well, which will print the error message and count the test as a failure.